### PR TITLE
PYTHON-3592 createEncryptedCollection should raise a specialized exception to report the intermediate encryptedFields

### PR DESCRIPTION
**Pull request by juliusgeo**

### CODE REVIEW

**Comment by juliusgeo** [comment]
Opened a ticket for the build failure: https://jira.mongodb.org/browse/PYTHON-3596

File: pymongo/encryption.py [hunk 1]
```diff
@@ -629,12 +630,7 @@ def create_encrypted_collection(
                         master_key=master_key,
                     )
                 except EncryptionError as exc:
-                    raise EncryptionError(
-                        Exception(
-                            "Error occurred while creating data key for field %s with encryptedFields=%s"
-                            % (field["path"], encrypted_fields)
-                        )
-                    ) from exc
+                    raise EncryptedFieldsError(exc, encrypted_fields) from exc
```
Comment (ShaneHarvey) [inline pymongo/encryption.py:633]: Can you update the docstring to mention that create_encrypted_collection raises this error, like this: https://github.com/mongodb/mongo-python-driver/blob/124dee66c3df47883b85df82b34f740ce35f69e1/pymongo/__init__.py#L164-L165

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: pymongo/encryption.py (hunk 1 above)
Comment (juliusgeo) [reply pymongo/encryption.py]: Done.

File: pymongo/errors.py [hunk 2]
```diff
@@ -359,6 +359,31 @@ def timeout(self) -> bool:
         return False
 
 
+class EncryptedFieldsError(EncryptionError):
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply pymongo/errors.py]: Done.

File: pymongo/errors.py [hunk 3]
```diff
@@ -359,6 +359,31 @@ def timeout(self) -> bool:
...
+
+    .. versionadded:: 4.4
+    """
+
+    def __init__(self, cause: Exception, encrypted_fields: Mapping[str, Any]) -> None:
+        super(EncryptedFieldsError, self).__init__(cause)
+        self.__encrypted_fields = encrypted_fields
+
+    @property
+    def encrypted_fields(self):
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply pymongo/errors.py]: Done.

**Review by blink1073 — APPROVED** [review]

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: pymongo/encryption.py [hunk 4]
```diff
@@ -614,6 +615,9 @@ def create_encrypted_collection(
         as keyword arguments to this method.
         See the documentation for :meth:`~pymongo.database.Database.create_collection` for all valid options.
 
+        :Raises:
+          - :py:class:`~pymongo.errors.EncryptedCollectionError`: When either data-key creation or creating the collection fails.
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply pymongo/encryption.py]: Done.

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**